packages/ic-response-verification/src/validation/v2_validation.rs

```diff
@@ -106,67 +91,46 @@ pub fn validate_expr_path(
     if path_might_exist_in_tree(&request_url_path, tree) {
         return Err(
             ResponseVerificationError::ExactExpressionPathMightExistInTree {
-                potential_expr_path: request_url_path.iter().map(|e| e.to_string()).collect(),
+                potential_expr_path: request_url_path
+                    .iter()
+                    .map(|e| String::from_utf8_lossy(e).to_string())
+                    .collect(),
                 provided_expr_path: expr_path.to_vec(),
                 request_path: request_path.to_string(),
             },
         );
     }
     request_url_path.pop(); // pop "<$>"
 
-    // if the expr_path matches the full URL with a wildcard
-    // there can't be a more precise path in the tree
-    request_url_path.push("<*>".into());
-    if original_path.eq(&request_url_path) {
```

Comment:
This code wasn't necessary, because we are anyway checking the full URL with a wildcard inside the loop